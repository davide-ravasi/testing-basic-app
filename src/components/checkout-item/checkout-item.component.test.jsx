import React from 'react';
import { shallow } from 'enzyme';
import { CheckoutItem } from './checkout-item.component';

describe('Checkout Item Component', () => {
    let wrapper, mockProps;
    const mockCartItems = {
        item: {
            imageUrl: 'https://i.ibb.co/bPmVXyP/black-converse.png',
            name: "Black Converse",
            price: 110,
            quantity: 1
        }
    };

    beforeEach(() => {
        mockProps = {
            cartItem: mockCartItems,
            clearItem: jest.fn(),
            addItem: jest.fn(),
            removeItem: jest.fn()
        };

        wrapper = shallow(<CheckoutItem {...mockProps} />);
    })

    it('render basic component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('call removeItem action', () => {
        wrapper.find('#remove-item').simulate('click');
        expect(mockProps.removeItem).toHaveBeenCalled();
    })
})