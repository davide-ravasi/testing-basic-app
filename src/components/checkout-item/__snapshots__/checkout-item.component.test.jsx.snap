// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Item Component render basic component 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="item"
    />
  </styled.div>
  <styled.span />
  <Styled(styled.span)>
    <div
      id="remove-item"
      onClick={[Function]}
    >
      ❮
    </div>
    <span />
    <div
      onClick={[Function]}
    >
      ❯
    </div>
  </Styled(styled.span)>
  <styled.span />
  <styled.div
    onClick={[Function]}
  >
    ✕
  </styled.div>
</styled.div>
`;
